### PR TITLE
move image tensor normalize under fluxion's utils

### PR DESCRIPTION
**Pull request by deltheil**

### CODE REVIEW

File: src/refiners/fluxion/utils.py [hunk 1]
```diff
@@ -34,6 +34,31 @@ def interpolate(x: Tensor, factor: float | torch.Size, mode: str = "nearest") ->
...
+
+    dtype = tensor.dtype
+
+    mean_tensor = as_tensor(mean, dtype=tensor.dtype, device=tensor.device)
+    std_tensor = as_tensor(std, dtype=tensor.dtype, device=tensor.device)
+
+    if (std_tensor == 0).any():
+        raise ValueError(f"std evaluated to zero after conversion to {dtype}, leading to division by zero.")
+
+    if mean_tensor.ndim == 1:
```
Comment (limiteinductive) [inline src/refiners/fluxion/utils.py:53]: isn't mean_tensor.ndim == 1 always true?

File: src/refiners/fluxion/utils.py [hunk 2]
```diff
@@ -34,6 +34,31 @@ def interpolate(x: Tensor, factor: float | torch.Size, mode: str = "nearest") ->
...
+        mean_tensor = mean_tensor.view(-1, 1, 1)
+
+    if std_tensor.ndim == 1:
```
Comment (limiteinductive) [inline src/refiners/fluxion/utils.py:56]: ditto

File: src/refiners/fluxion/utils.py [hunk 3]
```diff
@@ -34,6 +34,31 @@ def interpolate(x: Tensor, factor: float | torch.Size, mode: str = "nearest") ->
...
+def normalize(tensor: Tensor, mean: list[float], std: list[float], inplace: bool = False) -> Tensor:
+    assert tensor.is_floating_point()
+    assert tensor.ndim >= 3
+
+    if not inplace:
+        tensor = tensor.clone()
+
+    dtype = tensor.dtype
+
+    mean_tensor = as_tensor(mean, dtype=tensor.dtype, device=tensor.device)
```
Comment (limiteinductive) [inline src/refiners/fluxion/utils.py:47]: why using the as_tensor and not just the standard way to create a tensor? (data is a list of float here)

File: src/refiners/fluxion/utils.py [hunk 4]
```diff
@@ -34,6 +34,31 @@ def interpolate(x: Tensor, factor: float | torch.Size, mode: str = "nearest") ->
     )  # type: ignore
 
 
+# Adapted from https://github.com/pytorch/vision/blob/main/torchvision/transforms/_functional_tensor.py
+def normalize(tensor: Tensor, mean: list[float], std: list[float], inplace: bool = False) -> Tensor:
```
Comment (limiteinductive) [inline src/refiners/fluxion/utils.py:38]: maybe we can jaxtype annotate  (since it is in utils there is an ambiguity: this is a specific to image data

**Review by limiteinductive — APPROVED** [review]

**Review by catwell — APPROVED** [review]

**Review by limiteinductive — APPROVED** [review]

**Pull request merged**